TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/NettyJsonBodySerializeHandler.java

```diff
@@ -34,16 +34,17 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
         }
     }
 
-    private void serializePayload(ChannelHandlerContext ctx, Map<String,Object> payload) throws IOException {
-        var pac = jsonEmitter.getChunkAndContinuations(payload, NUM_BYTES_TO_ACCUMULATE_BEFORE_FIRING);
-        while (true) {
-            ctx.fireChannelRead(new DefaultHttpContent(pac.partialSerializedContents));
-            pac.partialSerializedContents.release();
-            if (pac.nextSupplier == null) {
-                ctx.fireChannelRead(LastHttpContent.EMPTY_LAST_CONTENT);
-                break;
+    private void serializePayload(ChannelHandlerContext ctx, Map<String, Object> payload) throws IOException {
+        try (var jsonEmitter = new JsonEmitter(ctx.alloc())) {
```

Comment:
Oh so this is why `JsonEmitter` implements `AutoCloseable`

Reply:
It implements AutoCloseable because it holds onto a CompositeByteBuf.  That needs to be released at some point and AutoCloseable makes that easy and self-documenting.  And this code here is where we take advantage of the auto-close trait.